config/initializers/secure_headers.rb

```diff
@@ -5,7 +5,6 @@
     config.ssl_options = { hsts: { preload: true, expires: 1.year, subdomains: true } }
 
     config.action_dispatch.default_headers.merge!(
-      'X-Frame-Options' => 'DENY',
```

Comment:
I expect this will (and should) raise eyebrows. Overriding the behavior of the Lookbook's controller isn't as easy as I'd hope it would be, despite the fact that [ViewComponent provides configurable preview controllers](https://viewcomponent.org/guide/previews.html#configuring-preview-controller) as an option. While it's likely possible to either monkey-patch it in or some other solution, I questioned whether the default `sameorigin` policy was actually insufficient, or if it would be fine to adopt it. I welcome others' thoughts!

Reply:
Do you have a patch of what the preview controller changes looked like? I'm happy to poke around and see if we can get it to work

Otherwise, I feel like we could add a small middleware that overrides the drops the X-Frame-Options header just for the `/components` routes

Reply:
Re: if `sameorigin` is sufficient or not... I have no idea, probably need to do some digging to see where we got the policy from

Reply:
Looks like we needed [a similar thing](https://github.com/18F/identity-idp/commit/a62bfb4234ba6fc4f4c46aead9d41d3fe2f20adb#diff-66cc7258886071348febbf60314501e02948b7d31e616fe88121ccc338ddb449R42) for mailer previews as well? Maybe we could hook into that logic, or fit it in to any updates here.

Reply:
I had the following as a controller implementation for ViewComponent, but based on [the compatibility implementation in Lookbook](https://github.com/allmarkedup/lookbook/blob/f4010d8b8f4939c5ae313daddc20b2cd6abc3648/app/controllers/lookbook/app_controller.rb#L173-L181), it's not called in the way I'd expect. Additionally, it doesn't look like there's a good way to fit in an equivalent implementation with Lookbook, without monkey-patching.

```rb
class ComponentsController < ViewComponentsController
  before_action :override_iframe_csp

  def override_iframe_csp
    if FeatureManagement.rails_csp_tooling_enabled?
      apply_secure_headers_override_with_rails_csp_tooling
    else
      apply_secure_headers_override_with_secure_headers
    end
  end

  def apply_secure_headers_override_with_secure_headers
    override_content_security_policy_directives(
      script_src: ["'sha256-BpP5cW0Gx+N8kLdqy2x9W5dfNFZ8S2v0wpHG3uysWyQ='"],
      iframe_src: ["'self'"],
    )
  end

  def apply_secure_headers_override_with_rails_csp_tooling
    policy = current_content_security_policy
    policy.iframe_src(["'self'"])
    policy.script_src(["'sha256-BpP5cW0Gx+N8kLdqy2x9W5dfNFZ8S2v0wpHG3uysWyQ='"])
    request.content_security_policy = policy
  end
end
```

(The `script_src` was to stop console logging about [this script](https://github.com/allmarkedup/lookbook/blob/f4010d8b8f4939c5ae313daddc20b2cd6abc3648/app/views/layouts/lookbook/app.html.erb#L14-L16))

Reply:
> Re: if `sameorigin` is sufficient or not... I have no idea, probably need to do some digging to see where we got the policy from

Looks like it came from #1091. Block-by-default is a sensible approach, though the default `sameorigin` still seems to be a reasonable defense to [the attack in question](https://cheatsheetseries.owasp.org/cheatsheets/Clickjacking_Defense_Cheat_Sheet.html).

>DENY, which prevents any domain from framing the content. The "DENY" setting is recommended unless a specific need has been identified for framing.

Happy to dig further to see if we can avoid it, or apply it in a limited context.

Reply:
Yeah `sameorigin` probably seems like the easiest solution overall 👍

Reply:
I was going to change this to do something like what we did with Rails mailers:

https://github.com/18F/identity-idp/blob/bb21d76182dd23e1757f64fd5b7905c5c39915b8/config/initializers/secure_headers.rb#L59-L63

But it caught my attention that this isn't being accounted for with the new `FeatureManagement.rails_csp_tooling_enabled?`, is that right? Will the mailer previews break if we switch?

Also, it seems that Rails default CSP allows for `:unsafe_inline` CSS, which _might_ be okay, but wasn't what we were allowing previously:

https://github.com/rails/rails/blob/01d836fcf384b2fa9c416705b0570737ac9ad7e0/railties/lib/rails/application_controller.rb#L10

Reply:
I missed that the config mostly moved to [content_security_policy.rb](https://github.com/18F/identity-idp/blob/main/config/initializers/content_security_policy.rb). It would be nice if we could make it a bit more specific to the routes we're concerned with, but followed the mailer pattern in 18bb81fc8.